### PR TITLE
Show TS on lock screen while navigating

### LINKED ISSUE #340: [Beacons] Keep phone unlocked while navigating
Similar to what Google Maps does when navigating (press power button to resume app).

### PR DESCRIPTION
**Pull request by qwerty287**
Should resolve #340

### CODE REVIEW

File: app/src/main/java/com/kylecorry/trail_sense/navigation/ui/NavigatorFragment.kt [hunk 1]
```diff
@@ -649,6 +651,23 @@ class NavigatorFragment : BoundFragment<ActivityNavigatorBinding>() {
         binding.location.text = formatService.formatLocation(gps.location)
 
         updateNavigationButton()
+
+        // show on lock screen
+        if (userPrefs.navigation.lockScreenPresence) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
```
Comment (kylecorry31) [inline app/src/main/java/com/kylecorry/trail_sense/navigation/ui/NavigatorFragment.kt:657]: Can the lock screen presence be turned off when you navigate away from the navigation tab? (i.e. it should only be active when you navigate to a beacon and lock your screen while on the navigation tab)

File: app/src/main/java/com/kylecorry/trail_sense/settings/NavigationSettingsFragment.kt [hunk 2]
```diff
@@ -69,6 +72,17 @@ class NavigationSettingsFragment : CustomPreferenceFragment() {
             true
         }
 
+        prefLockScreenPresense?.setOnPreferenceClickListener {
+            if (!prefs.navigation.lockScreenPresence) {
```
Comment (kylecorry31) [inline app/src/main/java/com/kylecorry/trail_sense/settings/NavigationSettingsFragment.kt:76]: If my other comment is implemented above, then this should no longer be needed

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1031,4 +1031,6 @@
     <string name="pref_flashlight_toggle_with_system_title">Toggle with system</string>
     <string name="pref_flashlight_toggle_with_system_summary">Displays a notification when the flashlight is turned on outside of Trail Sense</string>
     <string name="pref_flashlight_settings">pref_flashlight_settings</string>
+    <string name="pref_navigation_lock_screen_presence" translatable="false">pref_navigation_show_on_lock_screen</string>
+    <string name="pref_navigation_lock_screen_presence_title">Presence on lock screen while navigating</string>
```
Comment (kylecorry31) [inline app/src/main/res/values/strings.xml:1035]: Can this be renamed to "Keep unlocked while navigating"?

Comment (qwerty287) [reply app/src/main/res/values/strings.xml:1035]: Hmm it's not locked... 
I'm not sure, but "Keep unlocked while navigating" is strictly speaken not correct (but I can change it nevertheless, if you want)

File: app/src/main/java/com/kylecorry/trail_sense/navigation/ui/NavigatorFragment.kt (hunk 1 above)
Comment (qwerty287) [reply app/src/main/java/com/kylecorry/trail_sense/navigation/ui/NavigatorFragment.kt:667]: Fixed in https://github.com/kylecorry31/Trail-Sense/pull/677/commits/b5322cdef7fe649f81568cad582097802f26a4ca

File: app/src/main/java/com/kylecorry/trail_sense/settings/NavigationSettingsFragment.kt (hunk 2 above)
Comment (qwerty287) [reply app/src/main/java/com/kylecorry/trail_sense/settings/NavigationSettingsFragment.kt]: Fixed in https://github.com/kylecorry31/Trail-Sense/pull/677/commits/67b9d5c8d70a29c7902724371c58012f6c2802e8

**Comment by qwerty287** [comment]
Sorry for the changes of the imports of com.kylecorry.trail_sense.shared.* in https://github.com/kylecorry31/Trail-Sense/pull/677/commits/5ea6372385155bf3d6c0a5f188cd58b94baa1bf5, I used Android Studios "Optimize imports" feature

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (kylecorry31) [reply app/src/main/res/values/strings.xml:1035]: Okay, let's keep it like this for now, and I'll change it if I find a better term

**Comment by kylecorry31** [comment]
Looks good, thanks!

**Pull request merged**